acorn-walk/src/index.js

```diff
@@ -353,6 +353,8 @@ base.ExportNamedDeclaration = base.ExportDefaultDeclaration = (node, st, c) => {
   if (node.source) c(node.source, st, "Expression")
 }
 base.ExportAllDeclaration = (node, st, c) => {
+  if (node.exported)
+    c(node.exported, st, "Pattern")
```

Comment:
I don't think this should be walked as a pattern—it's a very different construct from patterns (doesn't bind locally, etc). Other exports also don't walk their identifiers as patterns.